### PR TITLE
Adding Devise

### PR DESCRIPTION
**Pull request by SnakyMz**
Things we have done:

- [ ] Remove current_user method in ApplicationController
- [ ] Install and Setup Devise
- [ ] Update views as per wireframe

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,12 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
```
Comment (Gambit142) [inline app/models/user.rb:5]: - [ ] Great work setting up all the modules for the devise gem. However, the essence of setting authentication in your application is to ensure that users authenticate who they are before having access to the application. This adds a layer of security and ensures that the user logging into the application is the actual person. When I run your application, I get direct access to the `Users#Index` page which shouldn't be. I had to manually input the URL for the sign-up page before creating a user. This should be done immediately after a user is directed to your root route. Kindly implement this and ensure that users don't have direct access to the application before authentication is done. 👍 🖱️ 

#### Hint: You can use the `  before_action :authenticate_user!` in the Users controller.

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @SnakyMz @Rysth , 

Good job so far!

## Highlights
- Setup linters correctly ✔️ 
- Use of devise for authentication ✔️ 
- Setup devise modules correctly ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app/models/user.rb (hunk 1 above)
Comment (SnakyMz) [reply app/models/user.rb:5]: Thank you very much.
I will implement it now

Comment (SnakyMz) [reply app/models/user.rb:5]: Thanks again for the suggestion
I have updated it as suggested

Screenshot 2023-11-09 002238

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @SnakyMz && @Rysth  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Made changes as requested by the previous reviewer :+1: 
- Project is working as expected :gear: 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (Gambit142) [reply app/models/user.rb:5]: > Thanks again for the suggestion I have updated it as suggested
> 
> Screenshot 2023-11-09 002238

You are welcome @SnakyMz

**Pull request merged**